Xamarin.Forms.Platform.WinRT/WindowsDeviceInfo.cs

```diff
@@ -72,43 +78,92 @@ public override double ScalingFactor
 			}
 		}
 
-		protected override void Dispose(bool disposing)
+		public override void BeginDeviceOrientationNotifications()
 		{
-			if (_isDisposed)
+			if (_simpleOrientationSensor != null)
 				return;
 
-			if (disposing)
-			{
-				_information.OrientationChanged -= OnOrientationChanged;
-				_information = null;
-			}
+			_simpleOrientationSensor = SimpleOrientationSensor.GetDefault();
+			_simpleOrientationSensor.OrientationChanged += OnDeviceOrientationChanged;
+		}
 
-			_isDisposed = true;
+		public override void EndDeviceOrientationNotifications()
+		{
+			if (_simpleOrientationSensor == null)
+				return;
 
-			base.Dispose(disposing);
+			_simpleOrientationSensor.OrientationChanged -= OnDeviceOrientationChanged;
+			_simpleOrientationSensor = null;
 		}
 
-		static DeviceOrientation GetDeviceOrientation(DisplayOrientations orientations)
+		void OnDeviceOrientationChanged(SimpleOrientationSensor sender, SimpleOrientationSensorOrientationChangedEventArgs args)
 		{
-			switch (orientations)
-			{
-				case DisplayOrientations.Landscape:
-				case DisplayOrientations.LandscapeFlipped:
-					return DeviceOrientation.Landscape;
+			SetDeviceOrientation(args.Orientation);
+		}
 
-				case DisplayOrientations.Portrait:
-				case DisplayOrientations.PortraitFlipped:
-					return DeviceOrientation.Portrait;
+		void OnScreenOrientationChanged(DisplayInformation sender, object args)
+		{
+			SetScreenOrientation(ApplicationView.GetForCurrentView().Orientation);
+		}
+
+		async void SetDeviceOrientation(SimpleOrientation orientation)
+		{
+			await Windows.ApplicationModel.Core.CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.High,
+			() =>
+			{
+				switch (orientation)
+				{
+					case SimpleOrientation.Rotated90DegreesCounterclockwise:
+						DeviceOrientation = _information.NativeOrientation == DisplayOrientations.Portrait ? DeviceOrientation.Landscape : DeviceOrientation.PortraitFlipped;
+						break;
+					case SimpleOrientation.Rotated180DegreesCounterclockwise:
+						DeviceOrientation = _information.NativeOrientation == DisplayOrientations.Portrait ? DeviceOrientation.PortraitFlipped : DeviceOrientation.LandscapeFlipped;
+						break;
+					case SimpleOrientation.Rotated270DegreesCounterclockwise:
+						DeviceOrientation = _information.NativeOrientation == DisplayOrientations.Portrait ? DeviceOrientation.LandscapeFlipped : DeviceOrientation.Portrait;
+						break;
+					case SimpleOrientation.NotRotated:
+						DeviceOrientation = _information.NativeOrientation == DisplayOrientations.Portrait ? DeviceOrientation.Portrait : DeviceOrientation.Landscape;
+						break;
+					default:
+						DeviceOrientation = DeviceOrientation.Unknown;
+						break;
+				}
+			}
```

Comment:
Check `NativeOrientation` to find default orientation on different form factors and adjust orientation based off of it and rotation degrees.